lib/hiera/backend/vault_backend.rb

```diff
@@ -31,7 +31,7 @@ def initialize()
             config.address = @config[:addr] if @config[:addr]
             config.token = @config[:token] if @config[:token]
             config.ssl_pem_file = @config[:ssl_pem_file] if @config[:ssl_pem_file]
-            config.ssl_verify = @config[:ssl_verify] if @config[:ssl_verify]
+            config.ssl_verify = @config[:ssl_verify] if config.respond_to? :ssl_verify
```

Comment:
I don't think this fix is correct.
The `vault-ruby` config will always `respond_to` `:ssl_verify`, see https://github.com/hashicorp/vault-ruby/blob/v0.4.0/lib/vault/configurable.rb#L20

A more correct fix would be:

```ruby
config.ssl_verify = @config[:ssl_verify] if @config.key?(:ssl_verify)
```

The intention behind this code is to only update the `config` if the option has been explicitly set in the `hiera.yml` config, otherwise leave the value as the default set by `vault-ruby`.

For some more context:
These `respond_to?` conditions were originally added as a safety while supporting older versions of vault-ruby (which may not support the specific SSL config option), however they can probably be removed now since this gem requires 0.4.